certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/service/GetCertificateEventsOfTypeDomainService.java

```diff
@@ -0,0 +1,146 @@
+package se.inera.intyg.certificateservice.domain.certificate.service;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import se.inera.intyg.certificateservice.domain.certificate.model.Certificate;
+import se.inera.intyg.certificateservice.domain.certificate.model.CertificateEvent;
+import se.inera.intyg.certificateservice.domain.certificate.model.CertificateEventType;
+import se.inera.intyg.certificateservice.domain.certificate.model.RelationType;
+import se.inera.intyg.certificateservice.domain.message.model.MessageType;
+
+public class GetCertificateEventsOfTypeDomainService {
+
+  public List<CertificateEvent> events(Certificate certificate, CertificateEventType type) {
+    switch (type) {
+      case CREATED -> {
+        return List.of(eventFromTimestamp(certificate, certificate.created(), type));
+      }
+      case SIGNED, AVAILABLE_FOR_PATIENT -> {
+        return List.of(eventFromTimestamp(certificate, certificate.signed(), type));
+      }
+      case SENT -> {
+        return certificate.sent() != null
+            ? List.of(eventFromTimestamp(certificate, certificate.sent().sentAt(), type))
+            : Collections.emptyList();
+      }
+      case INCOMING_MESSAGE -> {
+        return certificate.messages()
+            .stream()
+            .filter(message -> message.type() != MessageType.ANSWER
+                && message.type() != MessageType.COMPLEMENT)
+            .filter(message -> message.author().author().equals("FK"))
```

Comment:
Maybe the author class can be responsible for checking if its from FK?

```java
  private static final String FK = "FK";

  public boolean fromFK() {
    return this.name().equals(FK);
  }
```

Reply:
I can do that, but it's not gonna work with name since it's transformed to Försäkringskassan. I'll use author.